rclcpp/include/rclcpp/rate.hpp

```diff
@@ -112,8 +120,78 @@ class GenericRate : public RateBase
   std::chrono::time_point<Clock, ClockDurationNano> last_interval_;
 };
 
-using Rate = GenericRate<std::chrono::system_clock>;
-using WallRate = GenericRate<std::chrono::steady_clock>;
+class Rate : public RateBase
+{
+public:
+  RCLCPP_SMART_PTR_DEFINITIONS(Rate)
+
+  explicit Rate(
+    const double rate,
+    Clock::SharedPtr clock = std::make_shared<Clock>(RCL_SYSTEM_TIME))
+  : clock_(clock), period_(0, 0), last_interval_(clock_->now())
+  {
+    if (rate <= 0.0) {
+      throw std::invalid_argument{"rate must be greater than 0"};
+    }
+    period_ = Duration::from_seconds(1.0 / rate);
+  }
+  explicit Rate(
+    const Duration & period,
+    Clock::SharedPtr clock = std::make_shared<Clock>(RCL_SYSTEM_TIME))
+  : clock_(clock), period_(period), last_interval_(clock_->now())
+  {
+    if (period <= Duration(0, 0)) {
+      throw std::invalid_argument{"period must be greater than 0"};
+    }
+  }
```

Comment:
I think we should move these constructors into rate.cpp as well.

Reply:
Done in https://github.com/ros2/rclcpp/pull/2123/commits/a947972f50007dd3e0249f71f92017fcf22ca9e0